src/lib/kadm5/srv/svr_principal.c

```diff
@@ -616,16 +615,19 @@ kadm5_modify_principal(void *server_handle,
             free(adb.policy);
         adb.policy = strdup(entry->policy);
     }
-    if (have_pol) {
+
+    kdb->pw_expiration = 0;
```

Comment:
I think this has the new effect of unsetting the password expiration time whenever the policy isn't changed, for KDB modules other than LDAP.

Reply:
I don't think so?  At least, all I've done is hoisted the assignment out of the `else`.

Reply:
Agreed, fixed.